### PR TITLE
Update jackson to 2.9.10

### PR DESCRIPTION
**Pull request by ccaominh**
### Description

Addresses security vulnerabilities:
- sonatype-2016-0397: https://github.com/FasterXML/jackson-core/issues/315
- sonatype-2017-0355: https://github.com/FasterXML/jackson-core/pull/322

<hr>

This PR has:
- [x] been self-reviewed.
- [x] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/incubator-druid/blob/master/licenses.yaml)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by ccaominh** [comment]
Also manually tested using https://druid.apache.org/docs/latest/tutorials/tutorial-batch-hadoop.html

**Comment by jihoonson** [comment]
Would you please add a list of tests you have done?

**Comment by ccaominh** [comment]
In the test cluster, I ran a kinesis ingestion task to check the updated jackson version compatibility with AWS SDK for Java version 1.x.

@jihoonson Are there any additional manual tests that you recommend?

File: core/src/test/java/org/apache/druid/timeline/SegmentWithOvershadowedStatusTest.java [hunk 1]
```diff
@@ -74,42 +74,58 @@ public void testUnwrappedSegmentWithOvershadowedStatusDeserialization() throws E
...
+    Assert.assertEquals(dataSegment.getDimensions(), deserializedSegment.getDimensions());
+    Assert.assertEquals(dataSegment.getMetrics(), deserializedSegment.getMetrics());
+    Assert.assertEquals(dataSegment.getShardSpec(), deserializedSegment.getShardSpec());
+    Assert.assertEquals(dataSegment.getSize(), deserializedSegment.getSize());
+    Assert.assertEquals(dataSegment.getId(), deserializedSegment.getId());
+  }
 
+  // Previously, the implementation of SegmentWithOvershadowedStatus had @JsonCreator/@JsonProperty and @JsonUnwrapped
+  // on the same field (dataSegment), which used to work in Jackson 2.6, but does not work with Jackson 2.9:
+  // https://github.com/FasterXML/jackson-databind/issues/265#issuecomment-264344051
```
Comment (surekhasaharan) [inline core/src/test/java/org/apache/druid/timeline/SegmentWithOvershadowedStatusTest.java]: that's a bummer :(,  should we add a TODO note here to undo the workaround once https://github.com/FasterXML/jackson-databind/issues/1467 is fixed or when/if move to Jackson 3.x ?

Comment (ccaominh) [reply core/src/test/java/org/apache/druid/timeline/SegmentWithOvershadowedStatusTest.java]: Yeah, I can add a TODO. I think it'll work well in the implementation class as a reminder to remove the constructor I added.

**Comment by ccaominh** [comment]
Did another manual test that's a variant of https://druid.apache.org/docs/latest/tutorials/tutorial-batch-hadoop.html to have the hadoop index task ingest from s3 and then write to s3 via hdfs.

**Comment by ccaominh** [comment]
All the hadoop manual tests were done with hadoop 2.8.5.

**Pull request merged**